test/test_c10d.py

```diff
@@ -2370,6 +2370,57 @@ def check_no_grads():
         # No parameter should have their gradient set.
         check_no_grads()
 
+    @skip_if_not_multigpu
+    @skip_if_not_nccl
+    def test_accumulate_gradients(self):
+        gpus = gpus_for_rank(self.world_size)[self.rank][0:1]
+        self.assertEqual(len(gpus), 1)
+        store = c10d.FileStore(self.file.name, self.world_size)
+        process_group = c10d.ProcessGroupNCCL(store, self.rank, self.world_size)
+        local_batch_size = len(gpus)
+        global_batch_size = self.world_size * local_batch_size
+
+        model, ddp_model, input, target = \
+            self._prepare_single_device_module(
+                process_group, gpus, global_batch_size)
+
+        def step_model(model, input, target):
+            model.train()
+            output = model(input)
+            loss = F.mse_loss(output, target.to(output.device))
+            loss.backward()
+
+        # ensure accumulate grads works with no_grad
+        with torch.no_grad():
+            ddp_model.train()
+            ddp_model.module(input)
+
+        # check two model parameters over 2 iterations
+        for iteration in range(2):
```

Comment:
Let's do 4 iterations to make sure that things are still correct after the first sync.